src/Application.php

```diff
@@ -318,7 +320,15 @@ public function configureAndRegisterCommands(InputInterface $input, OutputInterf
         $discovery = $this->commandDiscovery();
         $commandClasses = $discovery->discover($commandfileSearchpath, '\Drush');
         $commandClasses[] = \Consolidation\Filter\Hooks\FilterHooks::class;
-        $commandClasses = array_merge($this->commandsFromConfiguration(), $commandClasses);
+        $commandsFromConfig = $this->commandsFromConfiguration();
+        if ($commandsFromConfig) {
+            @trigger_error("Discovering global commands by specifying them in 'drush.commands' config is deprecated and will be removed in Drush 11.0.0. See docs/commands.md on how to create auto-discoverable global commands.", E_USER_DEPRECATED);
+        }
+        $commandClasses = array_merge(
```

Comment:
The other question about this PR is, what happens if the same commandfile is found twice by multiple methods? For example, the [example-drush-extension](https://github.com/drush-ops/example-drush-extension) project creates an autoloadable class that is discovered based on its location, and is probably also discovered by this PR. I am presuming that this array_merge will cause these two identical items to become a single entry in the resulting array, since they should have the same key.

Reply:
Well, not all parts of the merge have keys. So, we should remove the keys > merge > array_unique(), assuming that there are no same two namespaces with different paths in this universe. That would be a nonsense and a bug somewhere.